app/models/item.js

```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+var Types = Schema.Types;
+var ObjectId = Types.ObjectId;
+
+var QueryPlugin = require('mongoose-query');
+
+var ItemSchema = new Schema({
+  barcode : { type: String, unique:true },
```

Comment:
Since barcode is no longer required we do not have to define it for items that do not have a barcode. Unique validation will only apply to those items that have a defined barcode.